docs/docs/clients/client-side/javascript.md

```diff
@@ -256,12 +256,14 @@ All function and property types can be seen
 | <code>getTrait(key:string)=> string&#124;number&#124;boolean</code>                                                                                 |                                                               Once used with an identified user you can get the value of any trait that is set for them e.g. `flagsmith.getTrait("accepted_cookie_policy")`                                                               |
 | <code>getAllTraits()=> Record&lt;string,string&#124;number&#124;boolean&gt;</code>                                                                  |                                                                      Once used with an identified user you can get a key value pair of all traits that are set for them e.g. `flagsmith.getTraits()`                                                                      |
 | <code>getState()=>IState</code>                                                                                                                     |                                                                               Retrieves the current state of flagsmith, useful in NextJS / isomorphic applications. `flagsmith.getState()`                                                                                |
+| <code>setState(state: IState)=>void</code>                                                                                                          |                                                                                                                    Set the current state of flagsmith.                                                                                                                    |
```

Comment:
Maybe this will be obvious to users of the JS client, but I think some kind of example here would be useful?

Reply:
This environmentID is the same used at the javascript-vanilla-example-v2 in the examples PR.

Reply:
I think perhaps mention that this is useful to hydration via SSR and link to https://docs.flagsmith.com/clients/next-ssr#comparing-ssr-and-client-side-flagsmith-usage, this is a more useful and full example. We'd rarely setState with just the environment id, more with flags/identity state.

Reply:
Maybe this way?
	
"Set the current state of flagsmith, [useful in NextJS / isomorphic applications.](https://docs.flagsmith.com/clients/next-ssr#comparing-ssr-and-client-side-flagsmith-usage) e.g. `flagsmith.setState({identity: 'mary@mycompany.com'})`."

Or should i give a different or bigger setState example? I thought of putting just one so the one-line code would not be ugly. Instead of keeping it in my thoughts, let's check:

`flagsmith.setState({identity: 'mary@mycompany.com', flags: {font_size: 12}})`.

What do you think?

Reply:
@kyle-ssg Should i do one of these two examples? I would like to close this PR before i get another one.

Reply:
@MatheusLasserre let's just go with the first option. As long as the link tot he relevant docs is there, I think that's good.

Reply:
Done!